invenio_rdm_records/services/config.py

```diff
@@ -21,92 +21,55 @@
     FileServiceConfig
 from invenio_records_resources.services.base.links import Link
 from invenio_records_resources.services.files.links import FileLink
-from invenio_records_resources.services.records.facets import \
-    NestedTermsFacet, RecordRelationLabels, TermsFacet
 from invenio_records_resources.services.records.links import RecordLink
-from invenio_vocabularies.services.facets import VocabularyLabels
 
 from ..records import RDMDraft, RDMRecord
-from ..records.systemfields.access.field.record import AccessStatusEnum
+from . import facets
 from .components import AccessComponent, ExternalPIDsComponent, \
     MetadataComponent, RelationsComponent
+from .customizations import FileConfigMixin, RecordConfigMixin, \
+    SearchOptionsMixin
 from .permissions import RDMRecordPermissionPolicy
 from .pids.providers import DOIDataCiteClient, DOIDataCitePIDProvider, \
     UnmanagedPIDProvider
 from .result_items import SecretLinkItem, SecretLinkList
 from .schemas import RDMParentSchema, RDMRecordSchema
 from .schemas.parent.access import SecretLink
 
-#
-# Facet definitions
-#
-resource_type_facet = NestedTermsFacet(
-    field='metadata.resource_type.props.type',
-    subfield='metadata.resource_type.props.subtype',
-    splitchar='::',
-    label=_("Resource types"),
-    value_labels=VocabularyLabels('resourcetypes')
-)
-
-language_facet = TermsFacet(
-    field='metadata.languages.id',
-    label=_('Languages'),
-    value_labels=VocabularyLabels('languages')
-)
-
-subject_facet = TermsFacet(
-    field='metadata.subjects.id',
-    label=_('Subjects'),
-    value_labels=VocabularyLabels('subjects')
-)
-
-access_status_facet = TermsFacet(
-    field='access.status',
-    label=_("Access status"),
-    value_labels={
-        AccessStatusEnum.OPEN.value: _("Open"),
-        AccessStatusEnum.EMBARGOED.value: _("Embargoed"),
-        AccessStatusEnum.RESTRICTED.value: _("Restricted"),
-        AccessStatusEnum.METADATA_ONLY.value: _("Metadata-only"),
-    }
-)
-
-is_published_facet = TermsFacet(
-    field='is_published',
-    label=_('State'),
-    value_labels={"true": _("Published"), "false": _("Unpublished")}
-)
-
 
 #
-# Search options
+# Default search configuration
 #
-class RDMSearchOptions(SearchOptions):
+class RDMSearchOptions(SearchOptions, SearchOptionsMixin):
```

Comment:
I keep the customize methods apart (in``customizations.py``), because I want a clean overview of the config.